kong/db/schema/others/declarative_config.lua

```diff
@@ -830,6 +838,20 @@ local function flatten(self, input)
         end
       end
 
+      if schema.ttl then
+        local expired_at = entry.ttl
+        if expired_at and expired_at ~= 0 and expired_at ~= null then
+          if entry.updated_at then
```

Comment:
Could it be:
```
flat_entry.ttl = expired_at + 
               entry.updated_at or entry.created_at or now_updated()

```

Reply:
have exported abosolute ttl, so we dont need to caculate expired_at time